tests/e2e/specs/shopper/filter-products-by-price.test.ts

```diff
@@ -223,4 +224,110 @@ describe( `${ block.name } Block`, () => {
 			);
 		} );
 	} );
+
+	describe( 'with Product Query Block', () => {
+		let editorPageUrl = '';
+		let frontedPageUrl = '';
+		beforeAll( async () => {
+			await switchUserToAdmin();
+			await createNewPost( {
+				postType: 'post',
+				title: block.name,
+			} );
+
+			await insertBlock( 'Product Query' );
+			await insertBlock( block.name );
+			await insertBlock( 'Active Product Filters' );
+			await page.waitForNetworkIdle();
+			await publishPost();
+
+			editorPageUrl = page.url();
+			frontedPageUrl = await page.evaluate( () =>
+				wp.data.select( 'core/editor' ).getPermalink()
+			);
+			await page.goto( frontedPageUrl );
+		} );
+
+		it( 'should render', async () => {
+			const products = await page.$$(
+				selectors.frontend.queryProductsList
+			);
+
+			expect( products ).toHaveLength( 5 );
+		} );
+
+		it( 'should show only products that match the filter', async () => {
+			const isRefreshed = jest.fn( () => void 0 );
+			page.on( 'load', isRefreshed );
+
+			await page.waitForSelector( block.class + '.is-loading', {
+				hidden: true,
+			} );
+
+			await expect( page ).toMatch( block.foundProduct );
+			expect( isRefreshed ).not.toBeCalled();
+
+			await Promise.all( [ setMaxPrice(), page.waitForNavigation() ] );
+
+			await page.waitForSelector( selectors.frontend.queryProductsList );
+			const products = await page.$$(
+				selectors.frontend.queryProductsList
+			);
+
+			const pageURL = page.url();
+			const parsedURL = new URL( pageURL );
+
+			expect( isRefreshed ).toBeCalledTimes( 1 );
+			expect( products ).toHaveLength( 1 );
+
+			expect( parsedURL.search ).toEqual(
+				block.urlSearchParamWhenFilterIsApplied
+			);
+			await expect( page ).toMatch( block.foundProduct );
+		} );
+
+		it( 'should refresh the page only if the user click on button', async () => {
+			await page.goto( editorPageUrl );
+
+			await openBlockEditorSettings( { isFSEEditor: false } );
+			await selectBlockByName( block.slug );
+			await page.waitForXPath(
+				block.selectors.editor.filterButtonToggle
+			);
+			const [ filterButtonToggle ] = await page.$x(
+				block.selectors.editor.filterButtonToggle
+			);
+			await filterButtonToggle.click();
+
+			await saveOrPublish();
+			await page.goto( frontedPageUrl );
+
+			const isRefreshed = jest.fn( () => void 0 );
+			page.on( 'load', isRefreshed );
+			await page.waitForSelector( block.class + '.is-loading', {
+				hidden: true,
+			} );
+			expect( isRefreshed ).not.toBeCalled();
+
+			await setMaxPrice();
```

Comment:
> I'm a bit confused by this. Why is setMaxPrice() after the assertion? Shouldn't it come before so we are really testing that changing the max price slider doesn't trigger a refresh?

This test is for the Product Query that, at this stage, when the filter is applied, requires the refresh of the page.

So, we are testing that the function is called only when the filter is applied and not before.

Reply:
Mmm, I see your point! That check was added because during the development of that block, there was a problem with the refresh, that the block, after rendered, refreshed the page. But I agree with you! I added another check [6d252af](https://github.com/woocommerce/woocommerce-blocks/pull/7351/commits/6d252af5cff89c783eeaac6562eb43f30e257122)